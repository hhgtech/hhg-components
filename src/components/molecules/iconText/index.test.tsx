import React from 'react'
import { render, cleanup, screen } from '@testing-library/react'
import 'jest-styled-components'

import { IconText } from '.'

import MockTheme from './../../../mockTheme'

afterEach(cleanup)

test('it renders', () => {
  const iconText = render(
    <MockTheme>
      <IconText
        size="lg"
        iconPosition="left"
        icon={<img src="https://source.unsplash.com/random/30x30" />}
      >
        Test Icon Text
      </IconText>
    </MockTheme>,
  )

  const self = iconText.container.firstChild

  expect(self).toMatchSnapshot()

  expect(self).toContainHTML('<img ')
  expect(self).toContainHTML('Test Icon Text')
  expect(self).toHaveAttribute('data-size', 'lg')
})
