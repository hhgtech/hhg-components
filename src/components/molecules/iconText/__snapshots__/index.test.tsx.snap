// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --font-weight-lg: 700;
  --font-weight-md: 400;
  --font-weight-sm: 400;
}

@media (max-width:767px) {
  .c0[data-size='lg'] {
    font-size: 16px;
    line-height: 24px;
    font-weight: var(--font-weight-lg);
  }

  .c0[data-size='md'] {
    font-size: 14px;
    line-height: 22px;
    font-weight: var(--font-weight-md);
  }

  .c0[data-size='sm'] {
    font-size: 13px;
    line-height: 20px;
    font-weight: var(----font-weight-sm);
  }
}

<div
  class="c0"
  data-size="lg"
>
  <img
    src="https://source.unsplash.com/random/30x30"
  />
  Test Icon Text
</div>
`;
